include/deal.II/sundials/arkode.h

```diff
@@ -0,0 +1,858 @@
+        prm.add_parameter("Absolute error tolerance", absolute_tolerance);
+        prm.add_parameter("Relative error tolerance", relative_tolerance);
+        prm.leave_subsection();
+      }
+
+      /**
+       * Initial time for the DAE.
+       */
+      double initial_time;
+
+      /**
+       * Final time.
+       */
+      double final_time;
+
+      /**
+       * Initial step size.
+       */
+      double initial_step_size;
+
+      /**
+       * Minimum step size.
+       */
+      double minimum_step_size;
+
+      /**
+       * Absolute error tolerance for adaptive time stepping.
+       */
+      double absolute_tolerance;
+
+      /**
+       * Relative error tolerance for adaptive time stepping.
+       */
+      double relative_tolerance;
+
+      /**
+       * Maximum order of ARK.
+       */
+      unsigned int maximum_order;
+
+      /**
+       * Time period between each output.
+       */
+      double output_period;
+
+      /**
+       * Maximum number of iterations for Newton or Fixed point method during
+       * time advancement.
+       */
+      unsigned int maximum_non_linear_iterations;
+
+      /**
+       * Specifies that the implicit portion of the problem is linear.
+       */
+      bool implicit_function_is_linear;
+
+      /**
+       * Specifies that the implicit portion of the problem is linear and time
+       * independent.
+       */
+      bool implicit_function_is_time_independent;
+    };
+
+    /**
+     * Constructor. It is possible to fine tune the SUNDIALS ARKode solver by
+     * passing an AdditionalData() object that sets all of the solver
+     * parameters.
+     *
+     * @param mpi_comm MPI communicator
+     * @param data ARKode configuration data
+     */
+    ARKode(const MPI_Comm mpi_comm = MPI_COMM_WORLD,
+           const AdditionalData &data=AdditionalData());
+
+    /**
+     * Destructor.
+     */
+    ~ARKode();
+
+    /**
+     * Integrate the initial value problem. This function returns the final
+     * number of computed steps.
+     */
+    unsigned int solve_ode(VectorType &solution);
+
+    /**
+     * Clear internal memory and start with clean objects. This function is
+     * called when the simulation start and when the user returns true to a
+     * call to solver_should_restart().
+     *
+     * By default solver_should_restart() returns false. If the user needs to
+     * implement, for example, local adaptivity in space, he or she may assign
+     * a different function to solver_should_restart() that performs all mesh
+     * changes, transfers the solution and the solution dot to the new mesh,
+     * and returns true.
+     *
+     * @param[in] t  The new starting time
+     * @param[in] h  The new starting time step
+     * @param[in,out] y   The new initial solution
+     */
+    void reset(const double &t,
+               const double &h,
+               const VectorType &y);
+
+    /**
+     * Reinit vector to have the right size, MPI communicator, etc.
```

Comment:
maybe "A function object that users need to supply and that is intended to reinit the given vector ..."

Same "A function  object that users need to supply and that..." language for the other objects below?